### PR TITLE
Add ARM64 support to the Containerfile example

### PR DESCRIPTION
**Pull request by nerdalert**
- Fixes ARM64 build for the containerfile
- Bump cuda:12.4.1-devel-ubi9 to the latest

### CODE REVIEW

**Comment by russellb** [comment]
#988 includes the `https` change, but applies it in a bunch more places, so I think we should apply that one first

**Comment by nerdalert** [comment]
@russellb rolled back the https change. Noticed another issue while looking at it. The order of the current containerfile won't use the GPU unless you swap the install order that I updated the PR with.

**Comment by russellb** [comment]
@nerdalert bonus points if you add some build automation for the image

**Review by russellb — APPROVED** [review]
lgtm, but looks like @cdoern wanted to delay merging it based on the milestone setting

**Comment by leseb** [comment]
@Mergifyio rebase

**Comment by leseb** [comment]
@nerdalert please rebase so we can move forward with this, thanks!
@cdoern are we good to go or do you still want to delay this?

**Comment by cdoern** [comment]
@leseb looks good, just rebase and this should be good to do! cc @nerdalert

**Comment by nerdalert** [comment]
@leseb @cdoern done! On rebase it picked up the x86 build arg `-mno-avx`. I just added a conditional for `aarch64`. Alternative is `Containerfile.ARM` but that's a UX decision. Be nice to still keep it simple imo. Double checked the build still works on a  Graviton ARM instance. I think I saw a big refactor in the pipe anyways. Thanks!

File: containers/cuda/Containerfile [hunk 1]
```diff
@@ -16,7 +15,12 @@ RUN export LD_LIBRARY_PATH="$LD_LIBRARY_PATH:/usr/local/cuda/lib64:/usr/local/cu
     && export XLA_FLAGS=--xla_gpu_cuda_data_dir=/usr/local/cuda
 
 ARG GIT_TAG=stable
-RUN CMAKE_ARGS="-DLLAMA_CUBLAS=on" CFLAGS="-mno-avx" python3.11 -m pip install -r https://raw.githubusercontent.com/instructlab/instructlab/${GIT_TAG}/requirements.txt --force-reinstall --no-cache-dir llama-cpp-python
+RUN if [ "$(uname -m)" = "aarch64" ]; then \
+        export CFLAGS=""; \
```
Comment (leseb) [inline containers/cuda/Containerfile:19]: Can't we just go with:

```bash
if [[ "$(uname -m)" != "aarch64" ]]; then
  export CFLAGS="-mno-avx"
fi
```

Or is `CFLAGS` always expected?j

**Review by leseb — COMMENTED** [review]
Small nit

File: containers/cuda/Containerfile (hunk 1 above)
Comment (nerdalert) [reply containers/cuda/Containerfile]: @leseb done! double checked on a graviton and amd64 instances.

File: containers/cuda/Containerfile [hunk 2]
```diff
@@ -1,10 +1,9 @@
 # SPDX-License-Identifier: Apache-2.0
 
-FROM nvcr.io/nvidia/cuda:12.3.2-devel-ubi9
-RUN dnf install -y python3.11 git python3-pip make automake gcc gcc-c++
+FROM nvcr.io/nvidia/cuda:12.4.1-devel-ubi9
+RUN dnf install -y python3.11 python3.11-devel git python3-pip make automake gcc gcc-c++
```
Comment (leseb) [inline containers/cuda/Containerfile:4]: Why do we need `python3.11-devel` all of the sudden?

**Review by leseb — APPROVED** [review]
Just left a small question but not a blocker, thanks!

File: containers/cuda/Containerfile (hunk 1 above)
Comment (leseb) [reply containers/cuda/Containerfile]: Thanks!

File: containers/cuda/Containerfile (hunk 2 above)
Comment (nerdalert) [reply containers/cuda/Containerfile:4]: @leseb  Good question, I should have mentioned it! aarch64 was wanting Python.h to build. So adding the headers with `python3.11-devel` fixes the dependency. Like a 300k package. ubi9 defaults to 3.9 so had to make it explicit to match paytohn3.11.

Comment (leseb) [reply containers/cuda/Containerfile:4]: Thanks for the clarification!

**Comment by leseb** [comment]
It's strange it looks like Mergify doesn't see the two approvals, can we merge this? @russellb @cdoern? Thanks!

**Comment by nathan-weinberg** [comment]
@Mergifyio rebase

**Review by nathan-weinberg — APPROVED** [review]

**Pull request merged**